api/dbadapters/bigquery.ts

```diff
@@ -134,12 +134,37 @@ export class BigQueryDbAdapter implements DbAdapter {
 
     if (metadata.location.toUpperCase() !== location.toUpperCase()) {
       throw new Error(
-        `Cannot create dataset "${schema}" in location "${location}". It already exists in location "${
-          metadata.location
-        }". Change your default dataset location or delete the existing dataset.`
+        `Cannot create dataset "${schema}" in location "${location}". It already exists in location "${metadata.location}". Change your default dataset location or delete the existing dataset.`
       );
     }
   }
+
+  private cleanRows(rows: any[]) {
+    if (rows.length === 0) {
+      return rows;
+    }
+
+    const sampleData = rows[0];
+    const BIGQUERY_DATE_CLASS_NAME = "BigQueryDate";
+    const fieldsWithBigQueryDates = Object.keys(sampleData).filter(
+      key =>
+        sampleData[key] &&
+        sampleData[key].constructor &&
+        sampleData[key].constructor.name === BIGQUERY_DATE_CLASS_NAME
+    );
+    if (fieldsWithBigQueryDates.length === 0) {
```

Comment:
suggest this which might be a little simpler:

```
fieldsWithBigQueryDates.forEach(dateField => {
  rows.forEach(row => row[dateField] = row[dateField].value);
});
```

Reply:
Fair enough, changed. I was vaguely thinking of not mutating the original variable but I don't think it matters too much.